hw/ip/otbn/doc/dv/index.md

```diff
@@ -530,203 +530,304 @@ The instruction-specific covergroup is `insn_loopi_cg`.
+  Tracked in `enc_bna_cg` as `wrd_XXXXXXXX_cross`, where `XXXXXXXX` is the base-2 representation of the bit being checked.
 
 #### BN.RSHI
 
 This instruction uses the `bnr` encoding schema, with covergroup `enc_bnr_cg`.
+There is no instruction-specific covergroup.
 
 No special coverage.
 
 #### BN.SEL
 
 This instruction uses the `bns` encoding schema, with covergroup `enc_bns_cg`.
+There is no instruction-specific covergroup.
 
 - Cross flag group, flag and flag value (2 times 4 times 2 points)
+  Tracked in `enc_bns_cg` as `flag_cross`.
 
 #### BN.CMP
 
 This instruction uses the `bnc` encoding schema, with covergroup `enc_bnc_cg`.
+There is no instruction-specific covergroup.
 
 - Extremal values of shift for both directions where the shifted value is nonzero
+  Tracked in `enc_bnc_cg` as `st_sb_nz_shifted_cross`.
 - A nonzero right shift with a value in `wrs2` whose top bit is set
+  Tracked in `enc_bnc_cg` as `srl_cross`.
 
 #### BN.CMPB
 
 This instruction uses the `bnc` encoding schema, with covergroup `enc_bnc_cg`.
+The instruction-specific covergroup is `insn_bn_subcmpb_cg`.
 
 - Extremal values of shift for both directions where the shifted value is nonzero
+  Tracked in `enc_bnc_cg` as `st_sb_nz_shifted_cross`.
 - A nonzero right shift with a value in `wrs2` whose top bit is set
+  Tracked in `enc_bnc_cg` as `srl_cross`.
 - Execute with both values of the carry flag for both flag groups (to make sure things are wired through properly)
+  Tracked as `fg_carry_flag_cross`.
 
 #### BN.LID
 
 This instruction uses the `bnxid` encoding schema, with covergroup `enc_bnxid_cg`.
+The instruction-specific covergroup is `insn_bn_xid_cg` (shared with `BN.SID`).
 
 - Load from a valid address, where `grs1` is above the top of memory and a negative `offset` brings the load address in range.
+  Tracked as `oob_base_neg_off_cross`.
 - Load from a valid address, where `grs1` is negative and a positive `offset` brings the load address in range.
+  Tracked as `neg_base_pos_off_cross`.
 - Load from address zero
+  Tracked as `addr0_cross`.
 - Load from the top word of memory
+  Tracked as `top_addr_cross`.
 - Load from an invalid address (aligned but above the top of memory)
-- Load from a misaligned address
+  Tracked as `oob_addr_cross`.
+- Misaligned address tracking.
+  Track loads from addresses that are in range for the size of the memory.
+  Crossing the possible misalignments for operand_a and offset would give a big cross (`32^2 = 1024`).
+  Instead, we split those alignments into 3 bins: 0, 1-30 and 31.
+  Crossing them gives 9 combinations, tracked as `part_align_cross`.
+  We also track all possible alignments of the sum as `addr_align_cross`.
 - See an invalid instruction with both increments specified
+  Tracked in `enc_bnxid_cg` as a bin of `incd_inc1_cross`.
 - See `grd` greater than 31, giving an illegal instruction error
+  Tracked as `bigb_cross`.
```

Comment:
Are you tracking cases where grs1_inc brings gsr1 out of bounds? Same for BN.SID.

Reply:
`grs1_inc` increments the register after the operation, not before. (In C, it would be `my_array[idx++]` rather than `my_array[++idx]`).

Reply:
To add to that: the increment operation always succeeds and doesn't have bounds. The bounds check is done *before* executing the instruction, i.e. if you have two consecutive BN.SID instructions, the first one can happily push grs1 out of bounds, and then the execution of the second one will fail.